File: 모범 코드/15일차/15. [필수] 출퇴근.java
import java.io.*;
import java.util.*;

public class main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        
        int N = Integer.parseInt(br.readLine());
        
        TreeSet<String> set = new TreeSet<>();
        for(int i=0; i<N; i++){
            st = new StringTokenizer(br.readLine());
            
            String name = st.nextToken();
            String status = st.nextToken();
            
            if(status.equals("enter")) set.add(name);
            else set.remove(name);
        }
        
        System.out.println(set.size());
        for(String name: set){
            System.out.println(name);
        }
    }
}

File: 모범 코드/15일차/15. [필수] 출퇴근.py
N = int(input())

s = set()
for i in range(N):
	name, status = input().split()
	
	if status == "enter":
		s.add(name)
	else:
		s.remove(name)
		
s = sorted(s)

print(len(s))
for name in s:
	print(name)